src/file_io.jl

```diff
@@ -920,6 +945,14 @@ function write_moments_data_to_binary(moments, fields, t, n_ion_species,
                               z, r, n_ion_species)
         append_to_dynamic_var(io_moments.thermal_speed, moments.charged.vth, t_idx, z, r,
                               n_ion_species)
+        if z.irank == 0 # lower wall 
+            append_to_dynamic_var(io_moments.chodura_integral_lower, moments.charged.chodura_integral_lower, t_idx, r,
+                              n_ion_species)
+        end
+        if z.irank == z.nrank - 1 # upper wall
+            append_to_dynamic_var(io_moments.chodura_integral_upper, moments.charged.chodura_integral_upper, t_idx, r,
+                              n_ion_species)
+        end
```

Comment:
This is a bit hacky, but should do the job until we implement a nicer, more generic mechanism for writing from only a subset of processes.

```suggestion
        if z.irank == 0 # lower wall 
            append_to_dynamic_var(io_moments.chodura_integral_lower, moments.charged.chodura_integral_lower, t_idx, r,
                              n_ion_species)
        else
            append_to_dynamic_var(io_moments.chodura_integral_lower, moments.charged.chodura_integral_lower, t_idx, 0,
                              n_ion_species)
        end
        if z.irank == z.nrank - 1 # upper wall
            append_to_dynamic_var(io_moments.chodura_integral_upper, moments.charged.chodura_integral_upper, t_idx, r,
                              n_ion_species)
        else
            append_to_dynamic_var(io_moments.chodura_integral_upper, moments.charged.chodura_integral_upper, t_idx, 0,
                              n_ion_species)
        end
```
Passing `0` instead of a `coordinate` uses the integer-handling mechanism (which was originally meant as a way to pass `n_species` instead of a coordinate), and means the range to write from in that dimension is `1:0`, which is empty. This should do what we want, because the processes in the `else` branch will call the HD5 write function (so avoiding a hang due to a collective operation on the I/O MPI communicator), but won't actually write anything.